modules/host/kernel/guest.config

```diff
@@ -0,0 +1,14 @@
+# Copyright 2022-2024 TII (SSRC) and the Ghaf contributors
+# SPDX-License-Identifier: CC-BY-SA-4.0
+#Below configurations needed for microvm kernel to boot
+CONFIG_SERIAL_8250=y
+CONFIG_SERIAL_8250_CONSOLE=y
```

Comment:
Where are `SERIAL_8250` needed? `-debug`-profiles? Or does something break if those are left out?

Reply:
If I don't include SERIAL_8250 then I see below error in journalctl
**ghaf-host microvm@kernel-vm[2140]: 2024/01/24 13:37:01 socat[2140] E UNIX-CLIENT: kernel-vm.sock: No such file or directory**
And microvm kernel won't boot at all.

Reply:
This is probably related to earlier fixed serial console support https://github.com/astro/microvm.nix/commit/ce0d2e393216e708d2abc228c68a7a8c2147aeec that @jpruiz84 got fixed as optional in microvm.nix just yesterday.

Let's keep it as open issue. For now, I'd say this is debug-feature and could be hardened if disabling it in microvm.nix declarations allows us to disable it in kernel config.

Reply:
Both are needed

CONFIG_SERIAL_8250=y
CONFIG_SERIAL_8250_CONSOLE=y

The Robot Framework checks if the netvm started correctly by checking the output of the serial console. Then, these modules are needed.

Reply:
Thanks. But they could be `-debug`-profile specific to support that testing requirement? 

In other words, there should be nothing blocking microvm.nix declared `netvm` from starting when serial is disabled and removed from kernel configs with `-release`-build where we won't enable that test automation interface?

Reply:
It is not recommended to test netvm boot process only in the debug profile. What we should do is to change the test procedure to make sure that netvm starts in debug and release profiles. I propose to change the test case ["Wait until NetVM service started" ](https://github.com/tiiuae/ci-test-automation/blob/ed45b93b8f42d338bccaec872ac2a860d4c4e1c6/Robot-Framework/resources/ssh_keywords.resource#L182)
from checking the systemctl status microvm@netvm output to check with ssh connection. @azbeleva